fms_fsdp/utils/checkpointing_utils.py

```diff
@@ -25,12 +25,8 @@ def get_latest(targdir, qualifier=lambda x: True):
     If directory is empty or nonexistent or no items qualify, return None."""
     if os.path.exists(targdir) and len(os.listdir(targdir)) > 0:
         latest = max(
-            [
-                os.path.join(targdir, x)
-                for x in os.listdir(targdir)
-                if qualifier(os.path.join(targdir, x))
-            ],
-            key=lambda path: int(path.split("/")[-1].split("_")[1]),
+            [x for x in os.listdir(targdir) if qualifier(os.path.join(targdir, x))],
+            key=lambda path: int(path.split("_")[1]),
```

Comment:
probably the same issue below with `get_oldest` ?

maybe we could add a unit test?

Reply:
Ok, removed `os.path.join` in the `get_oldest` function in commit 4a2b4b878f724e584efea980646a58c7286b470b, and added unit tests in commit f082164a6170df40f42f0469738fe302f303846e.

Reply:
looks better, though now the ctime key fails in get_oldest

Reply:
Ok, have pushed a fix, can you run the tests again to see if it works?